tidy3d/plugins/mode/mode_solver.py

```diff
@@ -348,6 +366,422 @@ def data_raw(self) -> ModeSolverData:
+                )
+                .rotated(theta, axis=self.bend_axis_3d)
+                .translated(x=translate_coords[0], y=translate_coords[1], z=translate_coords[2])
+            )
+
+            rotated_structures.append(structure.updated_copy(geometry=geometry))
+
+        return rotated_structures
+
+    @cached_property
+    def rotated_bend_center(self) -> List:
+        """Calculate the center at the rotated bend such that the modal plane is normal
+        to the azimuthal direction of the bend."""
+        rotated_bend_center = list(self.plane.center)
+        idx_rotate = 3 - self.normal_axis - self.bend_axis_3d
+        rotated_bend_center[idx_rotate] -= self.mode_spec.bend_radius
+
+        return rotated_bend_center
+
+    def _car_2_cyn(
+        self, mode_solver_data: ModeSolverData
+    ) -> Dict[Union[ScalarModeFieldCylindricalDataArray, ModeIndexDataArray]]:
+        """Convert cartesian fields to cylindrical fields centered at the
+        rotated bend center."""
+
+        # Extract coordinates from one of the six field components as they are colocated
+        pts = [mode_solver_data.Ex[name].values.copy() for name in ["x", "y", "z"]]
+        f, mode_index = mode_solver_data.Ex.f, mode_solver_data.Ex.mode_index
+
+        lateral_axis = 3 - self.normal_axis - self.bend_axis_3d
+
+        idx_w, idx_uv = self.plane.pop_axis((0, 1, 2), axis=self.bend_axis_3d)
+        idx_u, idx_v = idx_uv
+
+        pts[lateral_axis] -= self.rotated_bend_center[lateral_axis]
+        rho = np.sort(np.abs(pts[lateral_axis]))
+
+        theta = np.atleast_1d(
+            np.arctan2(
+                self.plane.center[idx_v] - self.rotated_bend_center[idx_v],
+                self.plane.center[idx_u] - self.rotated_bend_center[idx_u],
+            )
+        )
+        axial = pts[idx_w]
+
+        # Initialize output data arrays for cylindrical fields
+        field_data_cylindrical = {
+            field_name_cylindrical: np.zeros(
+                (len(rho), len(theta), len(axial), len(f), len(mode_index)), dtype=complex
+            )
+            for field_name_cylindrical in ("Er", "Etheta", "Eaxial", "Hr", "Htheta", "Haxial")
+        }
+
+        cos_theta, sin_theta = np.cos(theta), np.sin(theta)
+
+        axes = ("x", "y", "z")
+        axial_name, plane_names = self.plane.pop_axis(axes, axis=self.bend_axis_3d)
+        cmp_1, cmp_2 = plane_names
+        plane_name_normal = axes[self.normal_axis]
+        plane_name_lateral = axes[lateral_axis]
+
+        # Determine which coordinate transformation to use based on the lateral axis
+        if cmp_1 == plane_name_lateral:
+            lateral_coord_value = rho * cos_theta + self.rotated_bend_center[idx_u]
+        elif cmp_2 == plane_name_lateral:
+            lateral_coord_value = rho * sin_theta + self.rotated_bend_center[idx_v]
+
+        # Transform fields from cartesian to cylindrical
+        fields_at_point = {
```

Comment:
Guess this could be called something else now, like `fields_interp` or something.